google/cloud/spanner/internal/transaction_impl.h

```diff
@@ -68,8 +68,9 @@ class TransactionImpl {
   // If the TransactionSelector is in the "begin" state and the operation
   // successfully allocates a transaction ID, then the functor must assign
   // that ID to the selector. If the functor fails to allocate a transaction
-  // ID then it should place the selector into an error state. All of this
-  // is independent of whether the functor itself succeeds.
+  // ID then it should assign the status of the failing operation to the
```

Comment:
"the failing operation" may imply the (for example) `ExecuteSql()` instead of the `BeginTransaction()` that really failed to allocate the transaction ID.  I'm not sure how to tweak it though.